src/sys/socket/mod.rs

```diff
@@ -48,18 +48,31 @@ pub use self::multicast::{
     ip_mreq,
     ipv6_mreq,
 };
-pub use self::consts::*;
 
 pub use libc::sockaddr_storage;
 
 #[derive(Clone, Copy, PartialEq, Eq, Debug)]
 #[repr(i32)]
 pub enum SockType {
-    Stream = consts::SOCK_STREAM,
-    Datagram = consts::SOCK_DGRAM,
-    SeqPacket = consts::SOCK_SEQPACKET,
-    Raw = consts::SOCK_RAW,
-    Rdm = consts::SOCK_RDM,
+    Stream = libc::SOCK_STREAM,
+    Datagram = libc::SOCK_DGRAM,
+    SeqPacket = libc::SOCK_SEQPACKET,
+    Raw = libc::SOCK_RAW,
+    Rdm = libc::SOCK_RDM,
+}
+
+libc_enum!{
+    #[repr(i32)]
+    pub enum SockProtocol {
+        /// TCP protocol ([ip(7)](http://man7.org/linux/man-pages/man7/ip.7.html))
+        IPPROTO_TCP,
+        /// UDP protocol ([ip(7)](http://man7.org/linux/man-pages/man7/ip.7.html))
+        IPPROTO_UDP,
+        /// Allows applications to configure and control a KEXT
+        /// ([ref](https://developer.apple.com/library/content/documentation/Darwin/Conceptual/NKEConceptual/control/control.html))
+        #[cfg(any(target_os = "macos", target_os = "ios"))]
+        SYSPROTO_CONTROL,
+    }
 }
 
 // Extra flags - Supported by Linux 2.6.27, normalized on other platforms
```

Comment:
This should be switched to use `libc_bitflags` and the upstream declarations of these values.